### ISSUE TITLE
Improve website performance

### ISSUE THREAD

**Issue by Indigo744** [lyrgard/ffbeEquip#142]
Performance of the website (page loading speed, memory footprint, ...) could be improved.

Following PR already implements some improvements:
- #128 Optimize PNG files with PNGGauntlet
- #135 Optimize banner jpg image
- #138 Performance: lazyloading of units/equipments image

However, more could surely be done.

This thread is used to track performance enhancement work. Don't hesitate if you have any idea for improvements.

**Comment by Adezandee** [comment]
@Indigo744 
Yeah, I came up with the same idea some time ago but @lyrgard long term goal is to migrate to some kind of full blown JS frontend framework.
If you want to automate cache busting, I'll go for a Webpack configuration. This is kind of mandatory for advanced frontend framework anyway so this could be a first step

**Comment by Adezandee** [comment]
@Indigo744 
I'll go for a complete compilation of frontend assets (ie. static/ => /build/...). You can introduce all transformation steps necessary + a dedicated folder that could be directly served by Apache (bypassing express static middleware). This should be a considerable perf improvement.

**Comment by Indigo744** [comment]
I guess I could see how we could use Webpack to generate a build. Right now, it would only copy CSS/JS file, and maybe apply minimisation where possible. But still, how do you implements cache busting in this case? How do you reference these files?

**Comment by Adezandee** [comment]
I'll try something to do only the cache busting part. But this means that on every deployment, a compilation steps must be run ie `npm run compile`, are you ok with that @lyrgard ?

**Comment by Indigo744** [comment]
I think it should be as automated as possible, like the current automatic deployment you implemented, don't you think?

Also provide a way to automate build in development? I used to work with something like grunt-watch in the past. Not sure if it's still in use nowadays 😸 but you get what I mean.

**Comment by lyrgard** [comment]
@Adezandee  Oh, I'm sorry I didn't reply to your earlier question about templates ! I intended to, but didn't found the time at the moment, and to my shame I forgot...

I'm not again having to run npm run compile on deployment. Won't that hinder development a little too ? having to compile every change  before doing F5 in the browser. But I guess the server could be started in dev environment and serve files directly from static in that case, or something like that.

Another subject related to cache busting : in static/GL and static/JP, I have a dataVersion.json file with a number I increment every time I release a change on data, to clear the localStorage and force the new data to be downloaded. If we have a compile phase, we could replace it by a hash of the data files, don't you think ?

Also, minification of the js and json files would be great too !

**Comment by Darwe-Canine** [comment]
Can this be closed?

**Comment by lyrgard** [comment]
I'll close it in 2 days if no response
